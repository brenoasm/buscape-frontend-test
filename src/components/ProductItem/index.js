import React, { Fragment } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { colors } from "../../theme";

import ProductPreviewImageList from "../ProductPreviewImageList";
import ProductInformation from "../ProductInformation";

const propTypes = {
  product: PropTypes.object.isRequired,
  onBrokenImageUrl: PropTypes.func.isRequired,
  onSelectedImageChange: PropTypes.func,
  handleProductAddToCartClick: PropTypes.func
};

const defaultProps = {
  product: null,
  onBrokenImageUrl: () => {},
  onSelectedImageChange: () => {},
  handleProductAddToCartClick: () => {}
};

const StyledProductItem = styled.div`
  background-color: ${colors.white};
  margin: 15px 5px;
  padding: 20px;
  border-radius: 5px;
  box-shadow: 0 3px 3px ${colors.graySecondary};
  display: flex;
  justify-content: space-between;

  > div {
    flex: 2;
  }

  > div:last-child {
    margin-left: 20px;
  }
`;

const ProductItem = ({
  product,
  onBrokenImageUrl,
  onSelectedImageChange,
  handleProductAddToCartClick
}) => {
  return (
    <Fragment>
      {product && (
        <StyledProductItem>
          <ProductPreviewImageList
            selectedImage={product.selectedImage}
            images={product.images}
            productId={product.id}
            onBrokenImageUrl={onBrokenImageUrl}
            onSelectedImageChange={onSelectedImageChange}
          />
          <ProductInformation
            product={product}
            handleProductAddToCartClick={handleProductAddToCartClick}
          />
        </StyledProductItem>
      )}
    </Fragment>
  );
};

ProductItem.propTypes = propTypes;
ProductItem.defaultProps = defaultProps;

export default ProductItem;
