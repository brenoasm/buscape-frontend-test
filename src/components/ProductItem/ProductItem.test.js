import React from "react";
import { shallow } from "enzyme";

import ProductItem from "./";

describe("ProductItem", () => {
  const product = {
    id: 2321312,
    name: "Smartphone Apple iPhone 7 128GB",
    images: [
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
        selected: true
      },
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
        selected: false
      },
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
        selected: false
      },
      {
        url:
          "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
        selected: false
      }
    ],
    price: {
      value: 3509.1,
      installments: 10,
      installmentValue: 389.9
    },
    selectedImage:
      "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
    loading: false
  };

  it("matches the snapshot", () => {
    const wrapper = shallow(<ProductItem product={product} />);

    expect(wrapper).toMatchSnapshot();
  });

});
