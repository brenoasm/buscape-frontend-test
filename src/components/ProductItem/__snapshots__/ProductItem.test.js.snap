// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductItem
    handleProductAddToCartClick={[Function]}
    onBrokenImageUrl={[Function]}
    onSelectedImageChange={[Function]}
    product={
      Object {
        "id": 2321312,
        "images": Array [
          Object {
            "selected": true,
            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
          },
          Object {
            "selected": false,
            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
          },
          Object {
            "selected": false,
            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
          },
          Object {
            "selected": false,
            "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
          },
        ],
        "loading": false,
        "name": "Smartphone Apple iPhone 7 128GB",
        "price": Object {
          "installmentValue": 389.9,
          "installments": 10,
          "value": 3509.1,
        },
        "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <ProductPreviewImageList
          images={
            Array [
              Object {
                "selected": true,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
              },
              Object {
                "selected": false,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
              },
              Object {
                "selected": false,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
              },
              Object {
                "selected": false,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
              },
            ]
          }
          onBrokenImageUrl={[Function]}
          onSelectedImageChange={[Function]}
          productId={2321312}
          selectedImage="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
        />
        <ProductInformation
          handleProductAddToCartClick={[Function]}
          product={
            Object {
              "id": 2321312,
              "images": Array [
                Object {
                  "selected": true,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                },
              ],
              "loading": false,
              "name": "Smartphone Apple iPhone 7 128GB",
              "price": Object {
                "installmentValue": 389.9,
                "installments": 10,
                "value": 3509.1,
              },
              "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
            }
          }
        />
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ProductPreviewImageList
            images={
              Array [
                Object {
                  "selected": true,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                },
              ]
            }
            onBrokenImageUrl={[Function]}
            onSelectedImageChange={[Function]}
            productId={2321312}
            selectedImage="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
          />,
          <ProductInformation
            handleProductAddToCartClick={[Function]}
            product={
              Object {
                "id": 2321312,
                "images": Array [
                  Object {
                    "selected": true,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                  },
                ],
                "loading": false,
                "name": "Smartphone Apple iPhone 7 128GB",
                "price": Object {
                  "installmentValue": 389.9,
                  "installments": 10,
                  "value": 3509.1,
                },
                "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "images": Array [
              Object {
                "selected": true,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
              },
              Object {
                "selected": false,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
              },
              Object {
                "selected": false,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
              },
              Object {
                "selected": false,
                "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
              },
            ],
            "onBrokenImageUrl": [Function],
            "onSelectedImageChange": [Function],
            "productId": 2321312,
            "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleProductAddToCartClick": [Function],
            "product": Object {
              "id": 2321312,
              "images": Array [
                Object {
                  "selected": true,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                },
              ],
              "loading": false,
              "name": "Smartphone Apple iPhone 7 128GB",
              "price": Object {
                "installmentValue": 389.9,
                "installments": 10,
                "value": 3509.1,
              },
              "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-iwsKbI",
          "isStatic": true,
          "rules": Array [
            "
  background-color: ",
            "#FFF",
            ";
  margin: 15px 5px;
  padding: 20px;
  border-radius: 5px;
  box-shadow: 0 3px 3px ",
            "#D8D8D8",
            ";
  display: flex;
  justify-content: space-between;

  > div {
    flex: 2;
  }

  > div:last-child {
    margin-left: 20px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-iwsKbI",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <ProductPreviewImageList
            images={
              Array [
                Object {
                  "selected": true,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                },
              ]
            }
            onBrokenImageUrl={[Function]}
            onSelectedImageChange={[Function]}
            productId={2321312}
            selectedImage="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
          />
          <ProductInformation
            handleProductAddToCartClick={[Function]}
            product={
              Object {
                "id": 2321312,
                "images": Array [
                  Object {
                    "selected": true,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                  },
                ],
                "loading": false,
                "name": "Smartphone Apple iPhone 7 128GB",
                "price": Object {
                  "installmentValue": 389.9,
                  "installments": 10,
                  "value": 3509.1,
                },
                "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
              }
            }
          />
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ProductPreviewImageList
              images={
                Array [
                  Object {
                    "selected": true,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                  },
                ]
              }
              onBrokenImageUrl={[Function]}
              onSelectedImageChange={[Function]}
              productId={2321312}
              selectedImage="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
            />,
            <ProductInformation
              handleProductAddToCartClick={[Function]}
              product={
                Object {
                  "id": 2321312,
                  "images": Array [
                    Object {
                      "selected": true,
                      "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                    },
                    Object {
                      "selected": false,
                      "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                    },
                    Object {
                      "selected": false,
                      "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                    },
                    Object {
                      "selected": false,
                      "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                    },
                  ],
                  "loading": false,
                  "name": "Smartphone Apple iPhone 7 128GB",
                  "price": Object {
                    "installmentValue": 389.9,
                    "installments": 10,
                    "value": 3509.1,
                  },
                  "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "images": Array [
                Object {
                  "selected": true,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                },
                Object {
                  "selected": false,
                  "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                },
              ],
              "onBrokenImageUrl": [Function],
              "onSelectedImageChange": [Function],
              "productId": 2321312,
              "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleProductAddToCartClick": [Function],
              "product": Object {
                "id": 2321312,
                "images": Array [
                  Object {
                    "selected": true,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_2_c.jpg?v=2347575274",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_3_c.jpg?v=318433138",
                  },
                  Object {
                    "selected": false,
                    "url": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb/__200x400-PU98460_4_c.jpg?v=33273730",
                  },
                ],
                "loading": false,
                "name": "Smartphone Apple iPhone 7 128GB",
                "price": Object {
                  "installmentValue": 389.9,
                  "installments": 10,
                  "value": 3509.1,
                },
                "selectedImage": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-iwsKbI",
            "isStatic": true,
            "rules": Array [
              "
  background-color: ",
              "#FFF",
              ";
  margin: 15px 5px;
  padding: 20px;
  border-radius: 5px;
  box-shadow: 0 3px 3px ",
              "#D8D8D8",
              ";
  display: flex;
  justify-content: space-between;

  > div {
    flex: 2;
  }

  > div:last-child {
    margin-left: 20px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-iwsKbI",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
